### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by cgmillette**
hipTensor needs a docs overhaul to fit the new style, and hipTensor's new API and library dependencies:

* Update what-is-hiptensor
* Update installation.rst
* Update programmers-guide.rst
* Update api-reference.rst

Additional fixes include project configs and doxygen formatting in the code

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -32,13 +67,31 @@ GPU support
...
+hipTensor is designed to have minimal external dependencies such that it is light-weight and portable.
+
+.. <!-- spellcheck-disable -->
+
+* Minimum ROCm version support is 6.0.
+* Minimum cmake version support is 3.14.
+* Minimum ROCm-cmake version support is 0.8.0.
+* Minimum Composable Kernel version support is composable_kernel 1.1.0 for ROCm 6.0.2 (or ROCm package composablekernel-dev).
+* Minimum HIP runtime version support is 4.3.0 (or ROCm package ROCm hip-runtime-amd).
+* Minimum LLVM dev package version support is 10.0 (available as ROCm package rocm-llvm-dev).
```
Comment (deep-lurker) [inline docs/installation.rst]: We only check for llvm 7.0 and above in CMake

File: docs/installation.rst [hunk 2]
```diff
@@ -157,45 +257,203 @@ To build library and tests, run:
...
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. tabularcolumns::
+   |\X{2}{4}|\X{2}{4}|
+
+================================================ ===========================================================================================================================
+Executable name                                  Description
+================================================ ===========================================================================================================================
+``logger_test``                                  Unit test to validate hipTensor Logger APIs
+``yaml_test``                                    Unit test to validate the YAML functionality used to bundle and run test suites
+``bilinear_contraction_test_m2n2k2``             Bilinear contraction test [D = alpha * (A x B) + beta * C] with  half, single and mixed precision datatypes of rank 4
```
Comment (deep-lurker) [inline docs/installation.rst:280]: and m1n1k1

File: docs/installation.rst [hunk 3]
```diff
@@ -157,45 +257,203 @@ To build library and tests, run:
...
 
 .. code-block:: bash
 
-    cd docs
+    <test_exe> --gtest_filter=*name_filter*
 
-    sudo apt-get update
-    sudo apt-get install doxygen
-    sudo apt-get install texlive-latex-base texlive-latex-extra
+* Manually adjust the test cases coverage. Using your favorite text editor, you can modify test YAML configs to affect the test parameter coverage.
```
Comment (deep-lurker) [inline docs/installation.rst:440]: Should we specify somewhere that CK support only up to 2G elements in contraction?

File: docs/programmers-guide.rst [hunk 4]
```diff
@@ -27,49 +92,41 @@ The hipTensor code is split into four major parts:
...
+- ``02_permutation/permutation*``: Testing infrastructure for permutation tests.
+- ``02_permutation/rank*``: Testing harnesses for permutation of a particular rank.
+- ``02_permutation/configs``: YAML files with actual permutation testing parameters.
 
-- Jenkins is used to automate Continuous Integration (CI) testing.
+Contributing
+^^^^^^^^^^^^
 
-- ``clang-format`` is used to format C++ code.
+For those wishing to contribute to the project, please see `Contributing to hipTensor  <https://github.com/ROCm/hipTensor/blob/develop/CONTRIBUTING.md>`_.
```
Comment (deep-lurker) [inline docs/programmers-guide.rst]: dead link

File: docs/contributors-guide.rst [hunk 5]
```diff
@@ -104,3 +104,29 @@ Also, to install githooks to format the code per-commit, use:
...
+.githooks/install
+git checkout -b <new_branch>
+...
+git add <new_work>
+git commit -m "What was changed"
+git push origin <new_branch>
+...
+```
+
+<!-- markdownlint-disable ol-prefix -->
```
Comment (mkarunan) [inline docs/contributors-guide.rst]: contributors_error

Is this the intended behavior?

File: docs/programmers-guide.rst [hunk 6]
```diff
@@ -10,6 +10,71 @@ Programmer's guide
...
+* Tensor C/D modes [M0, ..., Mn, N0, ..., Nn]
+
+and repeated indices K0, ..., Kn are indices that are contracted. Contractions currently support up to M6N6K6 which means we may have up to 6 dimensions for each M, N and K.
+A tensor contraction with A = [M0, ..., M5, K0, ..., K5] B = [N0, ..., N5, K0, ..., K5] and C/D [M0, ..., M5, N0, ..., N5] would be considered a rank 12 contraction.
+
+Tensor mode
+^^^^^^^^^^^
+
+Tensor modes are a way for the user to easily specify the ordering or labeling of the input strides that define the dimensional data relationship. This is used to describe how
+the data is laid out in memory and how they related to each other spatially.
```
Comment (mkarunan) [inline docs/programmers-guide.rst]: how they are related to each other spatially.

File: docs/contributors-guide.rst [hunk 7]
```diff
@@ -104,3 +104,29 @@ Also, to install githooks to format the code per-commit, use:
 .. code-block:: bash
 
    ./.githooks/install
+
+Example
```
Comment (deep-lurker) [inline docs/contributors-guide.rst:108]: rendered content is not what we expect. 

image

**Review by deep-lurker — CHANGES_REQUESTED** [review]

**Comment by cgmillette** [comment]
Feedback has been incorporated

**Review by mkarunan — DISMISSED** [review]

**Review by CongMa13 — DISMISSED** [review]
lgtm

File: docs/contributors-guide.rst (hunk 5 above)
Comment (cgmillette) [reply docs/contributors-guide.rst]: Fixed

File: docs/programmers-guide.rst (hunk 6 above)
Comment (cgmillette) [reply docs/programmers-guide.rst]: fixed

**Review by CongMa13 — APPROVED** [review]
lgtm

**Pull request merged**